src/dune/file_tree.ml

```diff
@@ -517,6 +517,17 @@ end = struct
     in
     (Dir0.Contents.create ~files ~sub_dirs ~dune_file, dirs_visited)
 
+  let get_vcs ~default:vcs ~path ~readdir:{ Readdir.files; dirs } =
+    match
+      match
```

Comment:
Looks like this double `match` can be simplified away: just inline the result of the first one into the second.

Reply:
Ah, wait, I'm wrong, it's actually more like a bind. And also this code is apparently not new, so feel free to ignore this comment.